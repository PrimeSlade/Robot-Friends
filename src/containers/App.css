.text{
    font-weight: 600;
    color: #0ccac4;
}