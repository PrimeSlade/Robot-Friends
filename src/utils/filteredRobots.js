export const filteredRobots = (robots, searchField) => {
  return robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );
};
