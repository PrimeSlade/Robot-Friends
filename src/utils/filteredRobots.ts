import { Robot } from "../type/Types";

export const filteredRobots = (
  robots: Array<Robot>,
  searchField: string
): Array<Robot> => {
  return robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );
};
