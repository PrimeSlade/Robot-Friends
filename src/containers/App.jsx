import { robots } from "../robots";
import "./App.css";
import SearchBox from "../components/SearchBox";
import { lazy, Suspense, useState } from "react";
import Scroll from "../components/Scroll";

const CardList = lazy(() => import("../components/CardList"));

function App() {
  const [searchField, setSearchField] = useState(""); //defualt value is ""

  const filterdRobots = robots.filter(
    (user) => user.name.toLowerCase().includes(searchField) //since defualt is "" , every user has "" which will return all arrays
  );

  const [counter, setCounter] = useState(0);

  return (
    <>
      <div className="tc">
        <h1 className="text">RobotFriends</h1>
        <button onClick={() => setCounter((c) => c + 1)} className="mb3">
          Click {counter}
        </button>
        <SearchBox setSearchField={setSearchField} />
        <Scroll>
          <Suspense
            fallback={<div className="text mt3">Loading robots...</div>}
          >
            <CardList robots={filterdRobots} />
          </Suspense>
        </Scroll>
      </div>
    </>
  );
}

export default App;
