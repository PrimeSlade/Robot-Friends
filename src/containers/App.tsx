import { robots } from "../robots";
import "./App.css";
import SearchBox from "../components/SearchBox";
import { lazy, Suspense, useState } from "react";
import Scroll from "../components/Scroll";
import CounterButton from "../components/CounterButton";
import { filteredRobots } from "../utils/filteredRobots";
import React from "react";

const CardList = lazy(() => import("../components/CardList"));

function App() {
  const [searchField, setSearchField] = useState(""); //defualt value is ""

  const filterdRobots = filteredRobots(robots, searchField);

  return (
    <>
      <div className="tc">
        <h1 className="text">RobotFriends</h1>
        {/* <CounterButton /> */}
        <SearchBox setSearchField={setSearchField} />
        <Scroll>
          <Suspense
            fallback={<div className="text mt3">Loading robots...</div>}
          >
            <CardList robots={filterdRobots} />
          </Suspense>
        </Scroll>
      </div>
    </>
  );
}

export default App;
