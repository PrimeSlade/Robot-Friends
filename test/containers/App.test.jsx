import { render, waitFor } from "@testing-library/react";

import App from "../../src/containers/App";
import { filteredRobots } from "../../src/utils/filteredRobots";
import { robots } from "../../src/robots";

describe("App Component and utils tests", () => {
  test("testing app by using snapshot", async () => {
    let asFragment;
    await waitFor(async () => {
      const renderResult = render(<App />);
      asFragment = renderResult.asFragment;
    });
    expect(asFragment()).toMatchSnapshot();
  });

  test("testing filtered func", () => {
    const mockRobots = [
      { id: 1, name: "Robot One", email: "one@robots.com" },
      { id: 2, name: "Robot Two", email: "two@robots.com" },
      { id: 3, name: "Robot Three", email: "three@robots.com" },
    ];

    expect(filteredRobots(robots, "")).toEqual(robots);
    expect(filteredRobots(mockRobots, "one")).toEqual([mockRobots[0]]);
  });
});
