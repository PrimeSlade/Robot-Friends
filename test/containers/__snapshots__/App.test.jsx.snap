// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component and utils tests testing app by using snapshot 1`] = `
<DocumentFragment>
  <div
    class="tc"
  >
    <h1
      class="text"
    >
      RobotFriends
    </h1>
    <button
      class="mb3"
    >
      Click 0
    </button>
    <div>
      <input
        class="pa3 ba b--green bg-lightest-blue"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow: scroll; border: 1px solid black; height: 500px; margin-top: 50px;"
    >
      <div
        class="text mt3"
      >
        Loading robots...
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`testing app by using snapshot 1`] = `
<DocumentFragment>
  <div
    class="tc"
  >
    <h1
      class="text"
    >
      RobotFriends
    </h1>
    <button
      class="mb3"
    >
      Click 0
    </button>
    <div>
      <input
        class="pa3 ba b--green bg-lightest-blue"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow: scroll; border: 1px solid black; height: 500px; margin-top: 50px;"
    >
      <div
        class="text mt3"
      >
        Loading robots...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
